### ISSUE TITLE
[a11y]: AboutModal - Spacing between hyperlinks

### ISSUE THREAD

**Issue by embrown-ibm** [carbon-design-system/ibm-products#9571]
### Package

@carbon/ibm-products

### Browser

Chrome, Edge, FireFox

### Operating System

MacOS

### Package version

v2.79.0

### React version

v18.3.1

### Automated testing tool and ruleset

IBM Accessibility Checker

### Description

I expected the component to not have any a11y violations

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/docs/components-aboutmodal--overview

### Steps to reproduce

Using the IBM a11y Checker we just got several violations with the <AboutModal>.  When adding additional information to the modal such as copyright and licensing information, by default, the component only put 8 pixels between adjacent hyperlinks.  Evidently the guideline requires at least 12 pixels.
```
.c4p--about-modal .c4p--about-modal__links-container a + a {
    border-inline-start: 1px solid var(--cds-border-strong-01, #8d8d8d);
    margin-inline-start: 0.5rem;  <====== THE PROBLEM, one of these needs to be increase to at least .75rem
    padding-inline-start: 0.5rem; <====== THE PROBLEM, one of these needs to be increase to at least .75rem
}
```

<img width="3016" height="1020" alt="Image" src="https://github.com/user-attachments/assets/7c8cf328-4de8-40b3-9f2e-23af78c6d81c" />

<img width="1448" height="1360" alt="Image" src="https://github.com/user-attachments/assets/0467f035-8d3a-46e4-b1d9-e3aa67f17cf0" />

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

### Acceptance criteria
Revert the changes from https://github.com/carbon-design-system/ibm-products/pull/9702/changes
Increase the margin-inline-start from $spacing -03 to $spacing-04

**Comment by alina-jacob** [comment]
Since a PR will fix the Code asset via this issue, it'll be good to open separate Website and Figma issues to have those fixed respectively.
